pkgs/development/compilers/ponyc/default.nix

```diff
@@ -1,54 +1,85 @@
-{stdenv, glibc, fetchFromGitHub, llvm, makeWrapper, openssl, pcre2, coreutils }:
+{ stdenv, fetchFromGitHub, llvm, makeWrapper, pcre2, coreutils, which, libressl,
+  cc ? stdenv.cc, lto ? (if stdenv.isDarwin then false else true) }:
 
 stdenv.mkDerivation {
-  name = "ponyc-2016-07-26";
+  name = "ponyc-0.5.0";
 
   src = fetchFromGitHub {
     owner = "ponylang";
     repo = "ponyc";
-    rev = "4eec8a9b0d9936b2a0249bd17fd7a2caac6aaa9c";
-    sha256 = "184x2jivp7826i60rf0dpx0a9dg5rsj56dv0cll28as4nyqfmna2";
+    rev = "b3d7383399650a99e52bc3cf298c008f5b64a855";
+    sha256 = "1ykjaqhrsin908ls98gdg2dxprkf6zgdrfzgvfqvhp74srxg9pjr";
   };
 
-  buildInputs = [ llvm makeWrapper ];
+  buildInputs = [ llvm makeWrapper which ];
+  propagatedBuildInputs = [ cc ];
 
   # Disable problematic networking tests
   patches = [ ./disable-tests.patch ];
 
   preBuild = ''
     # Fix tests
     substituteInPlace packages/process/_test.pony \
-        --replace "/bin/cat" "${coreutils}/bin/cat"
+        --replace "/bin" "${coreutils}/bin"
```

Comment:
Adding `coreutils` to `PATH` or `buildInputs` doesn't help because `_test.pony` explicitly look for `/bin/echo` and `/bin/cat`.

I've updated the search string for the `substituteInPlace` to be rooted so that it will only match a `/bin` at the beginning of a path and not in the middle of the path.

Let me know if this still is too brittle or if you guys have any suggestions on how to handle this in a not so brittle manner.